homeassistant/components/renson/sensor.py

```diff
@@ -0,0 +1,310 @@
+    raw_format: bool
+
+
+@dataclass
+class RensonSensorEntityDescription(
+    SensorEntityDescription, RensonSensorEntityDescriptionMixin
+):
+    """Description of sensor."""
+
+
+SENSORS: tuple[RensonSensorEntityDescription, ...] = (
+    RensonSensorEntityDescription(
+        key="CO2_QUALITY_FIELD",
+        name="CO2 quality category",
+        field=CO2_QUALITY_FIELD,
+        raw_format=False,
+        state_class=None,
+    ),
+    RensonSensorEntityDescription(
+        key="AIR_QUALITY_FIELD",
+        name="Air quality category",
+        field=AIR_QUALITY_FIELD,
+        raw_format=False,
+        state_class=None,
+    ),
+    RensonSensorEntityDescription(
+        key="CO2_FIELD",
+        name="CO2 quality",
+        field=CO2_FIELD,
+        raw_format=True,
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.CO2,
+        native_unit_of_measurement=CONCENTRATION_PARTS_PER_CUBIC_METER,
+    ),
+    RensonSensorEntityDescription(
+        key="AIR_FIELD",
+        name="Air quality",
+        field=AIR_QUALITY_FIELD,
+        state_class=SensorStateClass.MEASUREMENT,
+        raw_format=True,
+        native_unit_of_measurement=CONCENTRATION_PARTS_PER_CUBIC_METER,
+    ),
+    RensonSensorEntityDescription(
+        key="CURRENT_LEVEL_FIELD",
+        name="Ventilation level",
+        state_class=None,
+        field=CURRENT_LEVEL_FIELD,
+        raw_format=True,
+    ),
+    RensonSensorEntityDescription(
+        key="CURRENT_AIRFLOW_EXTRACT_FIELD",
+        name="Total airflow out",
+        field=CURRENT_AIRFLOW_EXTRACT_FIELD,
+        raw_format=False,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement="m³/h",
+    ),
+    RensonSensorEntityDescription(
+        key="CURRENT_AIRFLOW_INGOING_FIELD",
+        name="Total airflow in",
+        field=CURRENT_AIRFLOW_INGOING_FIELD,
+        raw_format=False,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement="m³/h",
+    ),
+    RensonSensorEntityDescription(
+        key="OUTDOOR_TEMP_FIELD",
+        name="Outdoor air temperature",
+        field=OUTDOOR_TEMP_FIELD,
+        raw_format=False,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+    ),
+    RensonSensorEntityDescription(
+        key="INDOOR_TEMP_FIELD",
+        name="Extract air temperature",
+        field=INDOOR_TEMP_FIELD,
+        raw_format=False,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+    ),
+    RensonSensorEntityDescription(
+        key="FILTER_REMAIN_FIELD",
+        name="Filter change",
+        field=FILTER_REMAIN_FIELD,
+        raw_format=False,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement="days",
```

Comment:
https://github.com/home-assistant/core/blob/a6d6807dd0db5631e711cb24de1e52cb7a95d1cc/homeassistant/const.py#L595

Reply:
Please also use the duration device class, if it matches our criteria.

https://github.com/home-assistant/core/blob/a6d6807dd0db5631e711cb24de1e52cb7a95d1cc/homeassistant/components/sensor/const.py#L75

https://developers.home-assistant.io/docs/core/entity/sensor#available-device-classes